src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -171,6 +289,148 @@ QueryResult GetQueryResult(Document document, ref bool markedAsSkipped)
             ArrayPool<long>.Shared.Return(ids);
         }
 
+        private static int ProcessHighlightings(HighlightingField current, CoraxHighlightingTermIndex highlightingTerm, ReadOnlySpan<char> fieldFragment, List<string> fragments, int maxFragmentCount)
+        {
+            int totalFragments = 0;
+
+            // For each potential token we are looking for, and for each token that we need to find... we will test every analyzed token
+            // and decide if we create a highlightings fragment for it or not.
+            string[] values = (string[])highlightingTerm.Values;
+            for (int i = 0; i < values.Length; i++)
+            {
+                // We have reached the amount of fragments we required.
+                if (totalFragments >= maxFragmentCount)
+                    break;
+
+                var value = values[i];
+                var preTag = highlightingTerm.GetPreTagByIndex(i);
+                var postTag = highlightingTerm.GetPostTagByIndex(i);
+
+                int currentIndex = 0;
+                while (true)
+                {
+                    // We have reached the amount of fragments we required.
+                    if (totalFragments >= maxFragmentCount)
+                        break;
+
+                    // We found an exact match in the property value.
+                    var index = fieldFragment.Slice(currentIndex)
+                                             .IndexOf(value, StringComparison.InvariantCultureIgnoreCase);
+                    if (index < 0)
+                        break;
+
+                    index += currentIndex; // Adjusting to absolute positioning
+
+                    // We will look for a whitespace before the match to start the token. 
+                    int tokenStart = fieldFragment.Slice(0, index)
+                                                  .LastIndexOf(' ');
+                    if (tokenStart < 0)
+                        tokenStart = 0;
+
+                    // We will look for a whitespace after the match to end the token. 
+                    int tokenEnd = fieldFragment.Slice(index)
+                                                .IndexOf(' ');
+                    if (tokenEnd < 0)
+                        tokenEnd = fieldFragment.Length - index;
+
+                    tokenEnd += index; // Adjusting to absolute positioning
+
+                    int highlightingLength = preTag.Length + postTag.Length + (tokenEnd - tokenStart);
+
```

Comment:
There is an alternative interpretation here for the length. Should we include the size of the pre and post tag as part of the fragment size? cc @arekpalinski @ayende

Reply:
I don't think so.
It doesn't actually matter usually.